### PR TITLE
SAP HANA Express Quick Start Template

### PR DESCRIPTION
**Pull request by mszcool**
Re-Opening my previous pull request (#3458) with a new submission for SAP HANA Express Quick Start Template.

----------

Hi!
I am the Corp. DX TE for SAP and this quick-start-template is one for SAP HANA Express in Azure Virtual Machines. Given the requirement from SAP for customers to download the setup packages through their web site and accept the EULA, the quick-start template works as follows:
User is asked to register at SAP and download the HANA Express Setup Package from there.
After that, the script prepare-hxe-setup-files.sh creates a storage account with a container, uploads the HANA Express Setup Package and creates a Shared Access Signature URL that is valid for 1 year.
The user can then use this Shared Access Signature to deploy the Quick-Start through the portal or the included ARM template.
It would be great to prioritize the processing of this pull-request since we would love to present this during SAP Sapphire. I am also happy to accept any suggestions and process them as fast as possible before Tuesday, May 16th!
Thank you so much... and sorry for the short noticed (it was a short-notice project for me, as well;))
Cheers
Mario

### CODE REVIEW

**Comment by mszcool** [comment]
Ok, I really don't get this error. Can ANYONE HELP, please? the template works perfectly fine and just because of this Travis CI build not working it seems I cannot move forward and no one is looking at this!?

CAN ANYONE HELP CLARIFYING THIS ERROR? I ran the template as well as template validation commands and they raised no errors at all...

1) Template Running 1 Parallel Template Validation(s)... sap-hana-express:

      AssertionError: Template Validiation Failed. Try deploying your template with the commands:
azure group template validate --resource-group (your_group_name)  --template-file sap-hana-express/azuredeploy.json --parameters-file sap-hana-express/azuredeploy.parameters.json
azure group deployment create --resource-group (your_group_name)  --template-file sap-hana-express/azuredeploy.json --parameters-file sap-hana-express/azuredeploy.parameters.json 

Server Error:{
    "statusCode": 400,
    "body": {
        "error": "Error: EACCES: permission denied, open '68e77d3d-fcff-b100-d0a0-8e304bd2a6dd'"
    },
    "headers": {
        "x-powered-by": "Express",
        "content-type": "application/json; charset=utf-8",
        "content-length": "89",
        "etag": "W/\"59-0/KAFLdo/QRamQMZpNIIlA\"",
        "date": "Mon, 15 May 2017 18:07:16 GMT",
        "connection": "close"
    },
    "request": {
        "uri": {
            "protocol": "http:",
            "slashes": true,
            "auth": null,
            "host": "qstcliarmvalidatorapi.westus.cloudapp.azure.com:3000",
            "port": "3000",
            "hostname": "qstcliarmvalidatorapi.westus.cloudapp.azure.com",
            "hash": null,
            "search": null,
            "query": null,
            "pathname": "/validate",
            "path": "/validate",
            "href": "http://qstcliarmvalidatorapi.westus.cloudapp.azure.com:3000/validate"
        },
        "method": "post",
        "headers": {
            "Content-Type": "application/json",
            "content-length": 10785
        }
    }
}
      + expected - actual

**Comment by mszcool** [comment]
Ok, Travis CI/CD Build finally worked... waiting for what's next, then

**Comment by mszcool** [comment]
Is anyone picking this up or do I need to initiate the next step for this pull request to move forward?
Thank you very much
**Mario**

File: sap-hana-express/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+*.debug.parameters.json
```
Comment (bmoore-msft) [inline sap-hana-express/.gitignore]: remove this file from the PR

File: sap-hana-express/README.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    ![HANA Express Server Only Download Option](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/sap-hana-express/images/Figure03.png)
+
+4. Create a new Azure Resource Group, e.g. by using the Azure CLI as shown in the snippet below.
+
+    ``` bash
+    az group create --name "sampleresourcegroupname" --location "westeurope"
+    ```
+
+5. Execute the script "prep-hxe-setup-files.sh" shipping with this quick start template to upload the previously acquired (step 3 above) HANA Express Setup TAR archives to an Azure Storage Account and generate a Shared Access Signature for it:
```
Comment (bmoore-msft) [inline sap-hana-express/README.md]: the scripts supplied in the root of the repo will upload artifacts needed for deployment - let's use those scripts...

File: sap-hana-express/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+            "type": "string",
+            "minLength": 3,
+            "metadata": {
+                "description": "The basic hostname for the virtual machine. Will also be used to prefix the other resources related to the VM."
+            }
+        },
+        "vmSize": {
+            "type": "string",
+            "defaultValue": "Standard_DS5_v2",
+            "allowedValues": [
```
Comment (bmoore-msft) [inline sap-hana-express/azuredeploy.json]: remove allowed values, they won't work on all clouds

File: sap-hana-express/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+                "description": "The user name for the administrator/root user of the Virtual Machine deployed."
+            }
+        },
+        "adminPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "The password of the administrator user for being able to log into the provisioned virtual machine."
+            }
+        },
+        "hxeSetupFileUrl": {
```
Comment (bmoore-msft) [inline sap-hana-express/azuredeploy.json]: see BP document for handling artifacts

File: sap-hana-express/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+                "ipConfigurations": [
+                    {
+                        "name": "[concat(parameters('vmNamePrefix'), 'ipConfig1')]",
+                        "properties": {
+                            "privateIPAllocationMethod": "Dynamic",
+                            "publicIPAddress": {
+                                "id": "[resourceId('Microsoft.Network/publicIPAddresses', variables('vmPublicIpName'))]"
+                            },
+                            "subnet": {
+                                "id": "[concat(resourceId('Microsoft.Network/virtualNetworks', variables('vnetName')), '/subnets/', variables('vnetSubnetName'))]"
```
Comment (bmoore-msft) [inline sap-hana-express/azuredeploy.json]: simplify - just use resourceId

File: sap-hana-express/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,424 @@
...
+            "dependsOn": [
+                "[resourceId('Microsoft.Network/publicIPAddresses', variables('vmPublicIpName'))]",
+                "[resourceId('Microsoft.Network/virtualNetworks', variables('vnetName'))]",
+                "[resourceId('Microsoft.Network/networkSecurityGroups', variables('vnetNsgName'))]"
+            ]
+        },
+        {
+            "type": "Microsoft.Compute/disks",
+            "name": "[variables('vmDataDiskName')]",
+            "apiVersion": "2016-04-30-preview",
```
Comment (bmoore-msft) [inline sap-hana-express/azuredeploy.json]: update apiVersion to use a non-preview version

File: sap-hana-express/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,424 @@
...
+                    "createOption": "Empty"
+                },
+                "diskSizeGB": "[variables('vmDataDiskSizeInGb')]"
+            },
+            "dependsOn": []
+        },
+        {
+            "type": "Microsoft.Compute/virtualMachines",
+            "name": "[parameters('vmNamePrefix')]",
+            "apiVersion": "2016-04-30-preview",
```
Comment (bmoore-msft) [inline sap-hana-express/azuredeploy.json]: update apiVersion

File: sap-hana-express/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,424 @@
...
+            },
+            "dependsOn": [
+                "[resourceId('Microsoft.Compute/disks', variables('vmDataDiskName'))]",
+                "[resourceId('Microsoft.Network/networkInterfaces', variables('vmNicName'))]"
+            ],
+            "resources": [
+                {
+                    "type": "extensions",
+                    "name": "hxeinstallextension",
+                    "apiVersion": "2016-04-30-preview",
```
Comment (bmoore-msft) [inline sap-hana-express/azuredeploy.json]: updateAPI version

File: sap-hana-express/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,424 @@
...
+                        "[concat('Microsoft.Compute/virtualMachines/', parameters('vmNamePrefix'))]"
+                    ],
+                    "properties": {
+                        "publisher": "Microsoft.Azure.Extensions",
+                        "type": "CustomScript",
+                        "typeHandlerVersion": "2.0",
+                        "autoUpgradeMinorVersion": true,
+                        "settings": {
+                            "fileUris": [
+                                "[parameters('hxeInstallScriptUrl')]"
```
Comment (bmoore-msft) [inline sap-hana-express/azuredeploy.json]: see BP guide for handling artifacts

File: sap-hana-express/azuredeploy.parameters.json [hunk 10]
```diff
@@ -0,0 +1,24 @@
+{
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+        "vmNamePrefix": {
+            "value": "yourvmhostname"
```
Comment (bmoore-msft) [inline sap-hana-express/azuredeploy.parameters.json]: use GEN values where possible

File: sap-hana-express/scripts/prepare-hxe-setup-files.sh [hunk 11]
```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
```
Comment (bmoore-msft) [inline sap-hana-express/scripts/prepare-hxe-setup-files.sh]: remove this once updated to follow BP for artifacts

**Comment by bmoore-msft** [comment]
Closing abandoned request - please reopen if you can address the feedback.

**Pull request closed without merging**